gdx/src/com/badlogic/gdx/graphics/glutils/IndexBufferObject.java

```diff
@@ -56,33 +56,35 @@
 	boolean isBound = false;
 	final int usage;
 
+	// used to work around bug: https://android-review.googlesource.com/#/c/73175/
+	private final boolean empty;
+
+	/** Creates a new static IndexBufferObject to be used with vertex arrays.
+	 * 
+	 * @param maxIndices the maximum number of indices this buffer can hold */
+	public IndexBufferObject (int maxIndices) {
+		this(true, maxIndices);
+	}
+
 	/** Creates a new IndexBufferObject.
 	 * 
 	 * @param isStatic whether the index buffer is static
 	 * @param maxIndices the maximum number of indices this buffer can hold */
 	public IndexBufferObject (boolean isStatic, int maxIndices) {
-		byteBuffer = BufferUtils.newUnsafeByteBuffer(maxIndices * 2);
-		isDirect = true;
 
-		buffer = byteBuffer.asShortBuffer();
-		buffer.flip();
-		byteBuffer.flip();
-		bufferHandle = createBufferObject();
-		usage = isStatic ? GL20.GL_STATIC_DRAW : GL20.GL_DYNAMIC_DRAW;
-	}
+		empty = maxIndices == 0;
+		if (empty) {
+			maxIndices = 1; // avoid allocating a zero-sized buffer because of bug in Android's ART
+		}
 
-	/** Creates a new IndexBufferObject to be used with vertex arrays.
-	 * 
-	 * @param maxIndices the maximum number of indices this buffer can hold */
-	public IndexBufferObject (int maxIndices) {
 		byteBuffer = BufferUtils.newUnsafeByteBuffer(maxIndices * 2);
-		this.isDirect = true;
+		isDirect = true;
 
 		buffer = byteBuffer.asShortBuffer();
 		buffer.flip();
 		byteBuffer.flip();
 		bufferHandle = createBufferObject();
-		usage = GL20.GL_STATIC_DRAW;
+		usage = isStatic ? GL20.GL_STATIC_DRAW : GL20.GL_DYNAMIC_DRAW;

```

Comment:
Ah, ignore me, long day. Your fix is fine.
On Mar 24, 2015 6:20 PM, "Thorsten Schleinzer" <notifications@github.com>
wrote:

> In gdx/src/com/badlogic/gdx/graphics/glutils/IndexBufferObject.java
> <https://github.com/libgdx/libgdx/pull/1939#discussion_r27051160>:
>
> >
> >  		buffer = byteBuffer.asShortBuffer();
> >  		buffer.flip();
> >  		byteBuffer.flip();
> >  		bufferHandle = createBufferObject();
> > -		usage = GL20.GL_STATIC_DRAW;
> > +		usage = isStatic ? GL20.GL_STATIC_DRAW : GL20.GL_DYNAMIC_DRAW;
>
> Could explain why this is? The "isStatic" is always set and was ignored
> before my change. Or, putting it another way, how does one than ever switch
> to dynamic draw? Thank you! (Really don't want to bother you, just want to
> make sure to understand mistakes I make...)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/libgdx/libgdx/pull/1939/files#r27051160>.
>